### ISSUE TITLE
Moving towards bridge* RC

### ISSUE THREAD

**Issue by adriatic** [aurelia-ui-toolkits/issues#11]
### General info

I would like us all to focus on a single item for the next 7-10 days: getting KendoUI and Materialize bridge as close to Release Candidate as we can do, given the constraint that our code needs to run on Aurelia RC.

I propose to put @Thanood and @JeroenVinke is co-pilots and the rest of us to help with anything that two co-pilots request.

**Comment by adriatic** [comment]
### KendoUI Bridge tasks

_@adriatic's suggestions - to be validated and augmented by @JeroenVinke_ 
- [ ] All KendoUI books (installation, documentation) finished and installed (this happens in **[master](https://github.com/aurelia-ui-toolkits/aurelia-kendoui-bridge) branch**)
  assigned to @adriatic 
  
  _Note: installation and tutorials will be finished **after** Aurelia RC1 settles_.
- [ ] All variants of all samples resettled into gists (which branch? who does this?)
  _Jeroen might be the best person do do this since all samples are already stored as gists - just in a slightly different form, where I removed the "unique identifiers" in css tags, used to ensure that all css data is properly released_
- [ ] Upgrade from jspm 0.16.x to 0.17.x
- [x] Upgrade to Babel 6
- [ ] Remove core-js dependency
- [x] Upgrade to shadow dom

@JeroenVinke - please complete this list and create tasks [here](https://github.com/aurelia-ui-toolkits/aurelia-kendoui-bridge#boards?repos=44544444). **Do not try to do everything yourself and do not assign tasks to people who would take 10 times as much time as you would!** (this is a fine balancing act)

**Comment by adriatic** [comment]
### Materialize tasks

_@thanood, please do the similar list as the one above that I pre-created for @JeroenVinke. My advice is to wait until you see @JeroenVinke's list before starting on yours._

Edit Thanood:
- [x] release ShadowDOM changes
- [x] learn about Microsoft's current way on how to ship TypeScript definitions (referenced in `package.json`) and apply to the bridge ([reference](https://github.com/aurelia/framework/issues/360) and [another](https://blogs.msdn.microsoft.com/typescript/2016/06/15/the-future-of-declaration-files/) and [yet another](https://www.typescriptlang.org/docs/handbook/typings-for-npm-packages.html))
- [x] update the bridge's tutorial repo
- [x] finish [GitBook documentation](https://www.gitbook.com/book/aurelia-ui-toolkits/materialize-bridge-docs/details)
- [x] fix `md-card` problem
- [x] upgrade to Babel 6
- [ ] ~~upgrade to jspm 0.17~~ (postponed until Aurelia skeletons use that version, too)
- [x] remove core-js
- [ ] ~~fix jQuery 3 problem without explicitly introducing globals in `main.js`~~ (postponed due to Materialize not being compatible with jQuery 3), [possible reference](https://github.com/aurelia/framework/issues/445#issuecomment-227766890)
- [x] fix unit tests - karma crashes

**Comment by adriatic** [comment]
### Get control of Webpack

**This task will not start until having all webpack built skeletons working correctly**.

The [Using Webpack](https://github.com/aurelia-ui-toolkits/issues/issues/10) issue, that I wrote for Adam, once he stated his interest to handle all Aurelia-UI-Toolkits interactions with the webpack team. 

At this point, I am not sure about Adam's availability in the near future and I do not want to take any risks (like not having the webpack support for our bridges ready, despite the chance that the webkit would indeed work in the shape it is at the moment).

So, I declare myself in temporary charge of this task where I plan to
1. Collect all information about webpack that floated in chatrooms over last few days between  @charlespockert, @JeroenVinke and @Thanood).
2. Collect the most current information from @niieani (https://github.com/aurelia/skeleton-navigation/pull/440)
3. Run the verification decribed [here](https://github.com/aurelia-ui-toolkits/issues/issues/10#issuecomment-224775671)

**Comment by Thanood** [comment]
@adriatic The webpack tooling seems to have gone through a lot of changes recently. To be honest I have lost track of it. For example what exactly belongs in `aurelia.build.resources` in `package.json`?

I have been waiting for a working skeleton to refresh my knowledge but the current skeleton still has some problems. There are a lot of issues in `skeleton-navigation` and `webpack-plugin` repos about packages not being found. The whole `typings` thing (for typescript-webpack) seems to be in a state of transition.

I'd rather wait for a stable version of webpack plugin/loader/skeleton. I don't mean stable as in "release quality" but as in "things have settled and the team has decided which way to go with `aurelia.build` and typings config".

**Comment by adriatic** [comment]
@Thanood I agree with [your point(s) on webpack](https://github.com/aurelia-ui-toolkits/issues/issues/11#issuecomment-226091429) **completely**, so I corrected that entry to reflect that :-)

**Comment by niieani** [comment]
We'll post a blog post on the new Webpack support once that gets officially released. Stay tuned :)

**Comment by adriatic** [comment]
Thanks, @niieani for confirming that we (people that create Aurelia bridges) are indeed inserted in the proper look. We will happily wait for this blog post you mentioned and then make sure to verify that all problems with KendoUI bridge / Materialize bridge are taken care of.

**Comment by Thanood** [comment]
Regarding jspm 0.17.x: Is that version considered "stable" now?
Aurelia skeletons are still using 0.16.15

**Comment by niieani** [comment]
Nope, JSPM beta (0.17) is still in flux, at least in my tests it was broken.

**Comment by Thanood** [comment]
Good to know. Thanks, @niieani :smile:

**Comment by adriatic** [comment]
## @niieani can you see whether this is meaningful, please?

### Experiment with **[skeleton-webpack](https://github.com/aurelia/skeleton-navigation/tree/master/skeleton-webpack)**

"version": "1.0.0-beta.1.2.6"

#### Original app

The app **builds with a few errors** (not sure about significance of these errors, - see attached file 
**[pure-sample-build.txt](https://github.com/aurelia-ui-toolkits/issues/files/318837/pure-sample-build.txt)**.

The app **runs fine**

---

#### Modified app (added four KendoUI controls `autocomplete`, `button`, `chart` and `grid`)

This app fails to build. The attached file **[sample-with-kendoui-build.txt](https://github.com/aurelia-ui-toolkits/issues/files/319680/sample-with-kendoui-build.txt)** will hopefully help finding the problem.

My guesses:
1. This app is slightly modified original [skeleton-webpack](https://github.com/aurelia/skeleton-navigation/tree/master/skeleton-webpack), where I added the folder with all KendoUI files, it seems that my original problem remains:

```
...
[app.html] wants to require "kendo-ui/styles/kendo.common.min.css!", which does not exist.
[app.html] wants to require "kendo-ui/styles/kendo.bootstrap.min.css!", which does not exist.
...
```

I saw some comments that "css files cannot be `required`" so, this could be a problem. In addition, the attached "log" shows many repetitive statements like 

```
[kendoui/button/k-button.html] required "./kendoui/button/k-button.css" from "!!css!src/kendoui/button/k-button.css".
[kendoui/button/k-button.html] required "././kendoui/button/k-button.html" from "src/kendoui/button/k-button.html".
[kendoui/button/k-button.html] required "./kendoui/button/k-button.css" from "!!css!src/kendoui/button/k-button.css".
[kendoui/button/k-button.html] required "././kendoui/button/k-button.html" from "src/kendoui/button/k-button.html".
[kendoui/button/k-button.html] required "./kendoui/button/k-button.css" from "!!css!src/kendoui/button/k-button.css".
[kendoui/button/k-button.html] required "././kendoui/button/k-button.html" from "src/kendoui/button/k-button.html".
[kendoui/button/k-button.html] required "./kendoui/button/k-button.css" from "!!css!src/kendoui/button/k-button.css".
[kendoui/button/k-button.html] required "././kendoui/button/k-button.html" from "src/kendoui/button/k-button.html".
[kendoui/button/k-button.html] required "./kendoui/button/k-button.css" from "!!css!src/kendoui/button/k-button.css".
[kendoui/button/k-button.html] required "././kendoui/button/k-button.html" from "src/kendoui/button/k-button.html".
[kendoui/button/k-button.html] required "./kendoui/button/k-button.css" from "!!css!src/kendoui/button/k-button.css".
[kendoui/button/k-button.html] required "././kendoui/button/k-button.html" from "src/kendoui/button/k-button.html".
[kendoui/button/k-button.html] required "./kendoui/button/k-button.css" from "!!css!src/kendoui/button/k-button.css".
```

**Comment by niieani** [comment]
Repetitions are okay, they will be ignored in the actual build process. I'll look into the errors soon.

**Comment by adriatic** [comment]
@niieani Let me know which options you would prefer:
1. A description of the process that adds a KendoUI component to the [skeleton-webpack](https://github.com/aurelia/skeleton-navigation/tree/master/skeleton-webpack)
2. Have me create that app myself
3. None of the above

**Comment by Thanood** [comment]
If you asked me I'd say 1 and 2.. :smile:

**Comment by adriatic** [comment]
Not a difficult decision to make - so I will use you as Bazyli's proxy and provide 1 and 2 in this discussion.
